blog/2022-05-11-SDL-historization/2022-05-011-historization.md

```diff
@@ -0,0 +1,340 @@
+
+* initializing a MS SQL server
+* importing data into MS SQL table
+* injecting data into the data lake
+* modifying data on the SQL server side
+* re-copy / update data into data lake
+
+The data will be inspected and monitored using a Polynote notebook.
+
+
+## Test Case
+As a test case a [Chess Game Dataset](https://www.kaggle.com/datasets/datasnaek/chess) is selected. This data is a set of 20058 rows, in total 7MB. This is still faily small, but should be kind of representative.
+
+The dataset will be imported into the SQL server using the [db_init_chess.sql](db_init_chess.sql) script, which should be copied into the `config` directory. 
+
+It should be noted that there are a duplicates in the dataset. In the first case, the *deduplication* will be performed when the data is ported into the data lake (see configuration below). The procedure for table creation and modification is described below. 
+
+## Prerequisites
+
+* Podman installation
+* SDLB with metastore and Polynote by cloning the getting-started example: 
+  ```Bash
+  git clone https://github.com/smart-data-lake/getting-started.git SDL_sql
+	cd SDL_sql
+	unzip part2.additional-files.zip
+  ```
+* Utilizing JDBC to MS SQL server, SDLB required this additional dependency. Therefore, add the following dependency to the `pom.xml`:
+	```xml
+	<dependency>
+		<groupId>com.microsoft.sqlserver</groupId>
+		<artifactId>mssql-jdbc</artifactId>
+		<version>10.2.0.jre11</version>
+	</dependency>
+	```
+* build sdl-spark: `podman build -t sdl-spark .`
+* build the SDLB objects: 
+	```Bash
+	mkdir .mvnrepo
+	podman run -v ${PWD}:/mnt/project -v ${PWD}/.mvnrepo:/mnt/.mvnrepo maven:3.6.0-jdk-11-slim -- mvn -f /mnt/project/pom.xml "-Dmaven.repo.local=/mnt/.mvnrepo" package
+	```
+* copy polynote notebook [sql_data_monitor.ipynb](sql_data_monitor.ipynb) for later inspection into the `polynote/notebook` directory
+
+:::warning
+  The notebook will only be editable if the permissions are changed to be writable by other users `chmod +w polynote/notebook/sql_data_monitor.ipynb`
+:::
+
+* script for creating table on the SQL server: [db_init_chess.sql](db_init_chess.sql) into the `config` directory
+* script for modifying the table on the SQL server: [db_mod_chess.sql](db_mod_chess.sql) into the `config` directory
+* a restart script [restart_databases.sh](restart_databases.sh) is provided to clean and restart from scratch, including: stopping the containers, cleaning databases, freshly starting the containers and initializing the SQL database 
+
+## Prepare Source Database
+* start the pod with the metastore and polynote: 
+  ```Bash
+  podman-compose up
+  ```
+* start the MS SQL server: 
+  ```Bash
+  podman run -d --pod sdl_sql --hostname mssqlserver --add-host mssqlserver:127.0.0.1 --name mssql -v ${PWD}/data:/data  -v ${PWD}/config:/config -e "ACCEPT_EULA=Y" -e "SA_PASSWORD=%abcd1234%" mcr.microsoft.com/mssql/server:2017-latest
```

Comment:
I think it depends on the directory where you start podman-compose...
And i would leave it as "getting-started" because thats what is cloned by default. We should add a warning, and change "--pod sdl_sql" to "--pod getting-started" every where, wdyt?

Reply:
How? When cloning with the described:
`git clone https://github.com/smart-data-lake/getting-started.git SDL_sql` your directory schould be named SDL_sql and therewith also the pod sdl_sql. Did you checked out differently or did podman do something different in your case?

Reply:
Ok, i reused an already existing clone of getting-started, didn't see you checked out into a specific directory. Should we add a warning for that as well?

Reply:
Ok. Fair enough. I added a Note box for it.